packages/sdk/typescript/human-protocol-sdk/test/escrow.test.ts

```
@@ -0,0 +1,1368 @@
+        BigNumber.from(50),
+        BigNumber.from(50),
+        VALID_URL,
+        FAKE_HASH
+      );
+    });
+  });
+
+  describe('fund', () => {
+    test('should throw an error if escrowAddress is an invalid address', async () => {
+      const invalidAddress = FAKE_ADDRESS;
+      const amount = BigNumber.from(10);
+
+      await expect(escrowClient.fund(invalidAddress, amount)).rejects.toThrow(
+        ErrorInvalidEscrowAddressProvided
+      );
+    });
+
+    test('should throw an error if an hasEscrow return false', async () => {
+      const escrowAddress = ethers.constants.AddressZero;
+      const amount = BigNumber.from(10);
+
+      escrowClient.escrowFactoryContract.hasEscrow.mockReturnValue(false);
+
+      await expect(escrowClient.fund(escrowAddress, amount)).rejects.toThrow(
+        ErrorEscrowAddressIsNotProvidedByFactory
+      );
+    });
+
+    test('should throw an error if 0 <= amount', async () => {
+      const escrowAddress = ethers.constants.AddressZero;
+      const invalidAmount = BigNumber.from(0);
+
+      escrowClient.escrowFactoryContract.hasEscrow.mockReturnValue(true);
+
+      await expect(
+        escrowClient.fund(escrowAddress, invalidAmount)
+      ).rejects.toThrow(ErrorAmountMustBeGreaterThanZero);
+    });
+
+    test('should successfully fund escrow', async () => {
+      const tokenAddress = ethers.constants.AddressZero;
+      const escrowAddress = ethers.constants.AddressZero;
+      const amount = BigNumber.from(10);
+
+      escrowClient.escrowFactoryContract.hasEscrow.mockReturnValue(true);
+      escrowClient.escrowContract.token.mockReturnValue(tokenAddress);
+
+      await escrowClient.fund(escrowAddress, amount);
+
+      expect(escrowClient.escrowContract.token).toHaveBeenCalledWith();
+      expect(escrowClient.tokenContract.transfer).toHaveBeenCalledWith(
+        escrowAddress,
+        amount
+      );
+    });
+
+    test('should throw an error if the setup escrow fails', async () => {
```

fix writing:
'should throw an error if setup escrow fails'